File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/scheduler/scheduler.js
function solution(A) {
    var total = A.length;
    var answer = 0;
    var tick =0;
    var queue=[];

    while(true){
        var index = 0;
        while(true){
            if(index >= A.length) break;
            if(A[index][0] <= tick){
                queue.push(A.splice(index,1)[0]);
            }else{
                index++;
            }
        }

        if(queue.length===0) {
             if (A.length ===0) {
                 break;
             } else {
                tick++;
             }
        } else {
        var minIndex = 0;
        for(var i = 1 ; i < queue.length ; i++){
            if(queue[minIndex][1] > queue[i][1]) minIndex=i
        }
        tick += queue[minIndex][1];
        answer += tick - queue[minIndex][0];
        queue.splice(minIndex,1);

        }
    }

    return Math.floor(answer/total);
}

File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/scheduler/scheduler.py
def solution(A):
    total = len(A)
    answer = 0
    tick = 0
    q = []

    while True:
        index = 0
        while True:
            if index >= len(A): break
            if A[index][0] <= tick:
                q.append(A[index])
                A.remove(A[index])
            else:
                index += 1

        if len(q) == 0:
            if len(A) == 0: break
            else: tick += 1
        else:
            minIndex = 0
            for i in range(1, len(q)):
                if q[minIndex][1] > q[i][1]: minIndex = i
            tick += q[minIndex][1]
            answer += tick - q[minIndex][0]
            q.remove(q[minIndex])

    return answer / total